mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+        top_class_prob: NDArray,
+        y_pred: NDArray,
+        sample_weight: Optional[ArrayLike],
+    ) -> RegressorMixin:
+        """
+        Fitting the calibrator requires that we have a binary setting in place,
+        hence, we find the subset of values on which we want to apply this
+        calibration and apply a binary calibration on these.
+
+        Parameters
+        ----------
+        item : int
+            The class for which we fit a calibrator.
+        calibrator : RegressorMixin
+            Calibrator to fit.
+        y_calib : NDArray of shape (n_samples,)
+            Training labels.
+        top_class_prob : NDArray of shape (n_samples,)
+            The independent values of the calibrator, it represents the
+            maximum score of the predictions.
+        y_pred : NDArray of shape (n_samples,)
+            The array to determine which class the maximum prediction belongs
+            to.
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights. If ``None``, then samples are equally weighted.
+
+        Returns
+        -------
+        RegressorMixin
+            Calibrated estimator.
+        """
+        calibrator_ = clone(calibrator)
+        y_calib = cast(NDArray, y_calib)
+        sample_weight = cast(NDArray, sample_weight)
+        given_label_indices = np.where(y_pred.ravel() == item)[0]
+        y_calib_ = np.equal(y_calib[given_label_indices], item).astype(int)
+        top_class_prob_ = top_class_prob[given_label_indices]
+
+        if sample_weight is not None:
+            sample_weight_ = sample_weight[given_label_indices]
+            (
+                sample_weight_, top_class_prob_, y_calib_
+            ) = check_null_weight(  # type: ignore
+                sample_weight_,
+                top_class_prob_,
+                y_calib_
+            )
+        else:
+            sample_weight_ = sample_weight
+        calibrator_ = fit_estimator(
+            calibrator_, top_class_prob_, y_calib_, sample_weight_
+        )
+        return calibrator_
+
+    def _get_calibrator(
+        self,
+        calibrator: Optional[Union[str, RegressorMixin]],
+    ) -> RegressorMixin:
+        """
+        Check the input that has been provided for calibrator and
+        check that the calibrator is a valid estimator to calibrate.
+
+        Parameters
+        ----------
+        cv : Union[str, BaseCrossValidator]
+            Cross validation parameter.
+        calibrator : Union[str, RegressorMixin]
+            If calibrator is a string then it is linked to one of the
+            valid methods else calibrator as estimator.
+
+            By defaults ``None``. And if ``None``, we set to ``sigmoid``.
+
+        Returns
+        -------
+        RegressorMixin
+            RegressorMixin to be used as calibrator.
+        Raises
+        ------
+        ValueError
+            If calibrator is not one of the valid calibrators.
```

Comment:
valid is misleading. Maybe "if calibrator name is not a key of name_calibrators?
Again, you use calibrator in both sense, the method to calibrate and the calibrated model
I fear it might be misleading

Reply:
done.